airbyte-webapp/src/core/jsonSchema/schemaToYup.ts

```diff
@@ -109,6 +109,10 @@ export const buildYupFormForJsonSchema = (
     schema = schema.default(jsonSchema.default);
   }
 
+  if (schema && jsonSchema.enum) {
```

Comment:
This is actually handling the case that I described during the call, as storage is returned as an enum. In the case of GCP it's `GCP`. So that's how we try to understand what path is currently selected: validate based on json schema. 

So this piece of code just validates that only allowed values are selected for some property.